api/s2n.h

```diff
@@ -198,6 +199,7 @@ extern const uint8_t *s2n_connection_get_sct_list(struct s2n_connection *conn, u
 typedef enum { S2N_NOT_BLOCKED = 0, S2N_BLOCKED_ON_READ, S2N_BLOCKED_ON_WRITE } s2n_blocked_status;
 extern int s2n_negotiate(struct s2n_connection *conn, s2n_blocked_status *blocked);
 extern ssize_t s2n_send(struct s2n_connection *conn, const void *buf, ssize_t size, s2n_blocked_status *blocked);
+extern ssize_t s2n_sendv(struct s2n_connection *conn, const struct iovec *bufs, ssize_t count, ssize_t offs, s2n_blocked_status *blocked);
```

Comment:
Separately, what's the point of "offs"? writev(2) doesn't have an offset parameter, what's it do here? pwrite() does have an offset .. but it's for writing to the offset in a file. 

Helps to include a documentation change in the same PR :)

Reply:
Yeah, I'll add some documentation. 

The offs was a means for me to use the same iov them request is blocked. On s2n_send(), it's a simple matter of moving the pointer up but for iov, one needs to either modify existing one or create a copy of it and modify that. Nether of which is a bad choice in a write path. 

Offs was a means to use the same iov.

Reply:
My concern here is that it would violate the principal of least surprise for a caller who is used to writev() semantics. If writev() works a certain way and callers seem to be ok with it, is there really a problem? Seems like an IOV could be reused just by modifying it a little.

Reply:
Yeah, I will make one without offs as param. But I think there should be a separate one with offs though because if sendv was blocked, then the caller would have to create a new IOV before submitting it again. But the caller can receive the IOV as a constant object which cannot be modified in place and so a new one would have to be allocated. This is obviously not ideal. It would be better to pass the same data structure with offs field dictating how much to skip.